### ISSUE TITLE
No type error on interface implementation without return

### ISSUE THREAD

**Issue by SLaks** [google/closure-compiler#2158]
```js
/** @interface */
function Interface() {}

/** @return {number} */
Interface.prototype.x;

/** @implements {Interface } */
class Class {
  /** @override */
  x() {}
}
```

Even though the implementation has no `return` statement, the compiler doesn't complain that it doesn't return `number`.

If I add `return ''`, I do get an error.

https://closure-compiler-debugger.appspot.com/#input0%3D%252F**%2520%2540interface%2520*%252F%250Afunction%2520Interface()%2520%257B%257D%250A%250A%252F**%2520%2540return%2520%257Bnumber%257D%2520*%252F%250AInterface.prototype.x%253B%250A%250A%252F**%2520%2540implements%2520%257BInterface%2520%257D%2520*%252F%250Aclass%2520Class%2520%257B%250A%2520%2520%252F**%2520%2540override%2520*%252F%250A%2520%2520x()%2520%257B%257D%250A%257D%26input1%26conformanceConfig%26externs%26refasterjs-template%26includeDefaultExterns%3D1%26CHECK_LINT%3D1%26CHECK_SYMBOLS%3D1%26CHECK_TYPES%3D1%26CLOSURE_PASS%3D1%26MISSING_PROPERTIES%3D1%26PRESERVE_TYPE_ANNOTATIONS%3D1%26PRETTY_PRINT%3D1%26TRANSPILE%3D1

**Comment by dimvar** [comment]
For historical reasons, the missing return is checked by a separate pass. You need to enable `CHECK_MISSING_RETURN` in the debugger.

Again for historical reasons, the missing returns only fires for functions with non-empty bodies.

https://closure-compiler-debugger.appspot.com/#input0%3D%252F**%2520%2540interface%2520*%252F%250Afunction%2520Interface()%2520%257B%257D%250A%250A%252F**%2520%2540return%2520%257Bnumber%257D%2520*%252F%250AInterface.prototype.x%253B%250A%250A%252F**%2520%2540implements%2520%257BInterface%2520%257D%2520*%252F%250Aclass%2520Class%2520%257B%250A%2520%2520%252F**%2520%2540override%2520*%252F%250A%2520%2520x()%2520%257B%2520123%253B%2520%257D%250A%257D%26input1%26conformanceConfig%26externs%26refasterjs-template%26includeDefaultExterns%3D1%26CHECK_LINT%3D1%26CHECK_MISSING_RETURN%3D1%26CHECK_SYMBOLS%3D1%26CHECK_TYPES%3D1%26CLOSURE_PASS%3D1%26MISSING_PROPERTIES%3D1%26PRESERVE_TYPE_ANNOTATIONS%3D1%26PRETTY_PRINT%3D1%26TRANSPILE%3D1

IIRC, it is enabled by default if you have typechecking on.

NTI is better about this, and warns about missing returns without needing the extra pass, and even if the method has an empty body. We're unlikely to fix this in the old type checker.
